local-tests/chainrunner/node.py

```diff
@@ -0,0 +1,92 @@
+import json
+import os.path as op
+import re
+import requests
+import jsonrpcclient as rpc
+import subprocess
+
+from .utils import flags_from_dict
+
+class node:
+    """A class representing a single node of a running blockchain."""
+    def __init__(self, binary, chainspec, path, logdir=None):
+        self.chainspec = chainspec
+        self.binary = binary
+        self.path = path
+        self.logdir = logdir or path
+        self.logfile = None
+        self.flags = {}
+        self.running = False
+
+
+    def _stdargs(self):
+        return ['--base-path', self.path, '--chain', self.chainspec]
+
+
+    def start(self, name):
+        """Start the node. `name` is used to name of the logfile and for --name flag."""
+        cmd = [self.binary, '--name', name] + self._stdargs() + flags_from_dict(self.flags)
+        self.logfile = op.join(self.logdir, name+'.log')
```

Comment:
feel free to ignore this, probably we should have some linter set up, but I would wrap operators with spaces, e.g. `name + '.log'`

Reply:
I actually love that about Python that you can do operators both ways, with and without spaces ^^ That way you can decide which one to use to improve readability:
`y = 4 * x ^ 3 + 3 * x ^ 2 + 5 * x + 7`
`y = 4*x^3 + 3*x^2 + 5*x + 7`